ac/ac-timedQuiz/src/ActivityRunner.jsx

```diff
@@ -0,0 +1,209 @@
+// @flow
+
+import * as React from 'react';
+import { type ActivityRunnerT, TimedComponent } from 'frog-utils';
+import { ProgressBar, Button } from 'react-bootstrap';
+import { withState } from 'recompose';
+import { shuffle } from 'lodash';
+
+const styles = {
+  button: {
+    width: 'auto',
+    margin: '10px',
+    position: 'relative',
+    whiteSpace: 'normal',
+    float: 'center',
+    display: 'inline-block'
+  },
+  text: { width: '100%', fontSize: 'xx-large', textAlign: 'center' },
+  guidelines: { width: '100%' },
+  container: {
+    width: '500px',
+    height: '400px',
+    margin: 'auto',
+    marginTop: '80px'
+  },
+  main: {
+    width: '100%',
+    height: '100%',
+    position: 'absolute'
+  },
+  commands: {
+    width: '100%',
+    height: '50px',
+    margin: 'auto',
+    position: 'relative',
+    marginTop: '50px',
+    display: 'block'
+  }
+};
+
+const questionsWithIndex = props => {
+  const indexed = props.activityData.config.questions
+    .filter(q => q.question && q.answers)
+    .map((x, i) => [x, i]);
+  return indexed;
+};
+
+const shuffledQuestions = props => {
+  const questions = ['questions', 'both'].includes(
+    props.activityData.config.shuffle
+  )
+    ? shuffle(questionsWithIndex(props))
+    : questionsWithIndex(props);
+  return questions;
+};
+
+const generateExample = (q, progress) => {
+  const curQuestion = q[progress % q.length];
+
+  const startTime = Date.now();
+  return {
+    curQuestion,
+    startTime
+  };
+};
+
+let noAnswerTimeout;
+let delayTimeout;
+
+const Guidelines = ({ start, guidelines, name }) => (
+  <React.Fragment>
+    <div style={styles.text}>Welcome {name}!</div>
+    <div style={styles.guidelines}>{guidelines}</div>
+    <div style={{ ...styles.commands, width: '120px' }}>
+      <Button style={{ ...styles.button, width: '100%' }} onClick={start}>
+        {'Start'}
+      </Button>
+    </div>
+  </React.Fragment>
+);
+
+const CountDownTimer = TimedComponent(({ timeNow, length, start }) => {
+  const timeLeft = Math.ceil((length - Math.ceil(timeNow - start)) / 1000);
```

Comment:
why do you need to round with Math.ceil twice? Isn't the outer one enough?

Reply:
No, it ends up having weird skipping if there is not both when timeNow and start are certain values.